app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+import {useIntl} from 'react-intl';
+
+import {updateChannelNotifyProps} from '@actions/remote/channel';
+import OptionItem from '@components/option_item';
+import {useServerUrl} from '@context/server';
+import {preventDoubleTap} from '@utils/tap';
+
+type Props = {
+    channelId: string;
+    followedStatus: boolean;
+};
+
+const AutoFollowThreads = ({channelId, followedStatus}: Props) => {
+    const [autoFollow, setAutoFollow] = useState(followedStatus);
+    const serverUrl = useServerUrl();
+    const {formatMessage} = useIntl();
+
+    const toggleFollow = preventDoubleTap(() => {
+        const props: Partial<ChannelNotifyProps> = {
+            channel_auto_follow_threads: followedStatus ? 'off' : 'on',
+        };
+        setAutoFollow(!autoFollow);
+        updateChannelNotifyProps(serverUrl, channelId, props);
```

Comment:
what happens if this fails? I know is the same with other options, just want to see what would be the consequence and if we should also thing about this for the other options as well.

Reply:
Okay need to add checks for this and other option as well